MeetupPlatformApi/Features/Authentication/ChangeUserCredentials/ChangeUserCredentialsFeature.cs

```diff
@@ -0,0 +1,41 @@
+﻿namespace MeetupPlatformApi.Features.Authentication.ChangeUserCredentials;
+
+using BCrypt.Net;
+using MeetupPlatformApi.Persistence.Context;
+using MeetupPlatformApi.Seedwork.WebApi;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+[ApiSection(ApiSections.Authentication)]
+public class ChangeUserCredentialsFeature : FeatureBase
+{
+    private readonly ApplicationContext context;
+
+    public ChangeUserCredentialsFeature(ApplicationContext context)
+    {
+        this.context = context;
+    }
+
+    [HttpPut("/api/users/me/credentials")]
+    [Authorize]
+    public async Task<IActionResult> ChangeCredentials([FromBody] UserCredentialsChangeDto credentialsChangeDto)
+    {
+        var user = await context.Users.SingleOrDefaultAsync(user => user.Id == CurrentUser.UserId);
+        var unallowedUsername = credentialsChangeDto.Username == user.Username ? false
+            : await context.Users.AnyAsync(user => user.Username == credentialsChangeDto.Username);
+
+        if (user is null || !BCrypt.Verify(credentialsChangeDto.OldPassword, user.Password) || unallowedUsername)
+        {
+            return BadRequest($"User with username: {credentialsChangeDto.Username} is either fake or password is incorrect.");
```

Comment:
If user doesn't exist, then we should return a `404 Not Found`.
If user does exist, but the desired username is already taken by someone else, then we should return a `400 Bad Request`.

Reply:
Also, how a user or a username could be **"fake"**?